tensorflow/python/ops/custom_gradient.py

```diff
@@ -165,6 +165,90 @@ def second_order_and_transpose(ddy):
   Additional arguments to the inner `@tf.custom_gradient`-decorated function
   control the expected return values of the innermost function.
 
+  Above examples illustrates how you can create custom scalar gradient
+  function. Now, let us understand using linear polynomial on how to create
+  gradient which also computes gradient of a variable which can be `weights`
+  of model.
+
+  ```python
+  >>> trainable = []
+  ... # List to store the trainable variables (weights, biases etc.)
+  ...
+  ... @tf.custom_gradient
+  ... def linear_poly(x):
+  ...   # Weights will be trainable parameter and will represent coefficients
+  ...   # of poly and initialized as ones.
+  ...   weights = tf.Variable(tf.ones([2]))
+  ...   trainable.append(weights)
+  ...
+  ...   # Creating polynomial
+  ...   poly = weights[1] * x + weights[0]
+  ...
+  ...   def grad_fn(dpoly, variables=None):
+  ...     # dy/dx = weights[1] and we need to left multiply dpoly
+  ...     grad_xs = dpoly * weights[1]  # Scalar gradient
+  ...
+  ...     input_shape = len(x.shape)  # x contains how many dimensions
+  ...     grad_vars = []  # To store gradients of passed variables
+  ...     if variables is not None:
```

Comment:
done

Reply:
This still looks like it can be simplified: "assert variables is not None" and reduce the indent, no loop just "assert len(variables) == 1 and variables[0] is weights"